src/CometFlexVoting.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.10;
+
+import {Comet} from "comet/Comet.sol";
+import {CometConfiguration} from "comet/CometConfiguration.sol";
+import {Checkpoints} from "@openzeppelin/contracts/utils/Checkpoints.sol";
+
+import {FlexVotingClient} from "src/FlexVotingClient.sol";
+
+// TODO add description
+contract CometFlexVoting is Comet, FlexVotingClient {
+  using Checkpoints for Checkpoints.History;
+
+  /// @param _config The configuration struct for this Comet instance.
+  /// @param _governor The address of the flex-voting-compatible governance contract.
+  constructor(CometConfiguration.Configuration memory _config, address _governor)
+    Comet(_config)
+    FlexVotingClient(_governor)
+  {
+    _selfDelegate();
+  }
+
+  /// @notice Returns the current balance in storage for the `account`.
+  function _rawBalanceOf(address account) internal view override returns (uint256) {
+    int104 _principal = userBasic[account].principal;
+    return _principal > 0 ? uint256(int256(_principal)) : 0;
+  }
+
+  //===========================================================================
+  // BEGIN: Comet overrides
+  //===========================================================================
+  //
+  // This function is called any time the underlying balance is changed.
```

Comment:
I love how small our footprint of changes is to the contract. That's largely because of your awesome work abstracting the client contract, and then because of this assertion about the Comet contracts. I'll do a bit of poking around to sanity check it's true. Do you feel very confident about it. @mds1 did you get a chance to look at it?

Overall I think this is great. I agree with the approach and agree with seeing if we can get a PR into Compound's repo to get the function marked as virtual.

Reply:
Thank you!

> Do you feel very confident about it

Yeah, I'm fairly certain this function is the only way the principal balance is ever changed. The balance is stored as [`userBasic[account].principal`](https://github.com/compound-finance/comet/blob/3780c06b4eaa80a8c78e0ff770a7e8a1518db75e/contracts/CometStorage.sol#L68-L69). If you search through Comet.sol and CometExt.sol for `principal =` or `userBasic[.*=` you find that the only place where the principal can change is [here](https://github.com/compound-finance/comet/blob/3780c06b4eaa80a8c78e0ff770a7e8a1518db75e/contracts/Comet.sol#L759), in `updateBasePrincipal`.

I'll open my PR with them and see if they disagree.

Reply:
https://github.com/compound-finance/comet/pull/749

Reply:
I have not looked closely at coment, but I can spend some time reviewing to confirm the assumption if you want

Reply:
Up to you guys. We can also just wait for the Compound eng people to get back to us

Reply:
I took a look and I agree with your assessment @davidlaprade. Combined with the fact that we have the PR with Compound I think we can close this.